biorbd_optim/plot.py

```diff
@@ -161,16 +161,23 @@ def __init__(self, ocp, sol):
         self.sol = sol
 
     def show_biorbd_viz(self, show_meshes=True):
-        if self.ocp.nlp[0]["problem_type"] == ProblemType.torque_driven:
-            x, _, _ = ProblemType.get_data_from_V(self.ocp, self.sol["x"])
-
-        if self.ocp.nlp[0]["problem_type"] == ProblemType.muscles_and_torque_driven:
-            x, _, _, _ = ProblemType.get_data_from_V(self.ocp, self.sol["x"])
+        x = ProblemType.get_q_from_V(self.ocp, self.sol["x"])
 
         if self.ocp.nb_phases == 1:
             x = [x]
-        # x = self.ocp.nlp[0]["q_mapping"].expand(x)
 
+        else:
+            same_dof = True
+            for k in range(self.ocp.nlp[0]["model"].nbDof()):
+                if self.ocp.nlp[0]["model"].nameDof()[k].to_string() != self.ocp.nlp[1]["model"].nameDof()[k].to_string():
+                    same_dof = False
+                    break
+            if same_dof:
+                x_concat = x[0]
+                for i in range (1, self.ocp.nb_phases):
+                    x_concat = np.concatenate((x_concat, x[i]), axis = 1)
+                x = [x_concat]
+            
         from BiorbdViz import BiorbdViz
 
         try:
```

Comment:
In any case, the "import BiorbdViz" should be in the try... Actually, it should be the only thing in the try. We could argue that it is probably the only line that should be in the try...

Reply:
NOTE, I cannot comment the line later, but there is a "save_npy" function. The name is not explicit enough. We don't want to save numpy, we want to save the results. 
Still, using np.save is an unsatifactory way of saving according to me. All ocp and sol should be saved..

Reply:
However, an export_results_numpy would make sense